src/main/java/duke/util/List.java

```diff
@@ -0,0 +1,149 @@
+        System.out.println(newToDo);
+        System.out.println("Now you have " + list.size() + " tasks in the list.");
+        store.writeToFile(list);
+    }
+
+    /**
+     * Add Deadline Task to list.
+     *
+     * @param text Description of Deadline.
+     * @param by Deadline of Deadline Task.
+     */
+    public void addDeadline(String text, String by) {
+        try {
+            Task newDl = new Deadline(text, by);
+            list.add(newDl);
+            System.out.println("Got it. I've added this duke.task: ");
+            System.out.println(newDl);
+            System.out.println("Now you have " + list.size() + " tasks in the list.");
+            store.writeToFile(list);
+        } catch (DukeException | DateTimeParseException e) {
+            System.out.println("Please input your date and time in dd/mm/yyyy hhmm format, eg. 28/08/2021 2359");
+        }
+
+    }
+
+    /**
+     * Add Event Task to list.
+     *
+     * @param text Description of Event Task.
+     * @param at Date time of Event.
+     */
+    public void addEvent(String text, String at) {
+        try {
+            Task newEvent = new Event(text, at);
+            list.add(newEvent);
+            System.out.println("Got it. I've added this duke.task: ");
+            System.out.println(newEvent);
+            System.out.println("Now you have " + list.size() + " tasks in the list.");
+            store.writeToFile(list);
+        } catch (DukeException | DateTimeParseException e) {
+            System.out.println("Please input your date and time in dd/mm/yyyy hhmm format, eg. 28/08/2021 2359");
+        }
+    }
+
+    /**
+     * Set Task to be done at list index.
+     *
+     * @param index Position of Task to be set to done.
+     */
+    public void setIndexDone(int index) {// starts from 1
+        if (index > list.size() || index < 1) {//check for invalid index number
+            System.out.println("There is no duke.task " + index);
+            return;
+        }
+        list.get(index - 1).setDone();
+        System.out.println("Nice! I've marked this duke.task as done: ");
+        System.out.println(list.get(index - 1).toString());
+        store.writeToFile(list);
+    }
+
+    /**
+     * Delete Task from list.
+     *
+     * @param index Index of task to be deleted in list.
+     */
+    public void deleteTask(int index) {//starts from 1
+        if (index > list.size() || index < 1) {
+            System.out.println("There is no duke.task " + index);
+            return;
+        }
+        System.out.println("Noted. I've removed this duke.task: ");
+        System.out.println(list.get(index - 1).toString());
+        list.remove(index - 1);
+        System.out.println("Now you have " + list.size() + " tasks in the list.");
+        store.writeToFile(list);
+    }
+
+    /**
+     * Find Tasks with the input String in it.
+     *
+     * @param input Keyword to search Tasks with.
+     */
+    public void find(String input) {
```

Comment:
Shouldn't method names be a little more descriptive? Perhaps consider naming the method something that will tell us what it does (i.e. findTask)!

> A name is not just for differentiation; it should explain the named entity to the reader accurately and at a sufficient level of detail.

Reply:
nice catch thanks